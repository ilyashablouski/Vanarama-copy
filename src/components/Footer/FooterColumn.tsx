import React, { FC } from 'react';
import dynamic from 'next/dynamic';
import ReactMarkdown from 'react-markdown';
import LogoFacebookV2 from 'core/assets/icons/LogoFacebookV2';
import Icon from 'core/atoms/icon/Icon';
import LogoTwitterV2 from 'core/assets/icons/LogoTwitterV2';
import LogoInstagramV2 from 'core/assets/icons/LogoInstagramV2';
import LogoLinkedinV2 from 'core/assets/icons/LogoLinkedinV2';
import LogoYouTubeV2 from 'core/assets/icons/LogoYouTubeV2';
import Logo from 'core/atoms/logo';
import { PrimaryFooter_primaryFooter_linkGroups as LinkGroups } from '../../../generated/PrimaryFooter';
import RouterLink from '../RouterLink/RouterLink';
import { LinkTypes } from '../../models/enum/LinkTypes';
import Skeleton from '../Skeleton';

const Text = dynamic(() => import('core/atoms/text'), {
  loading: () => <Skeleton count={1} />,
});
const Heading = dynamic(() => import('core/atoms/heading'), {
  loading: () => <Skeleton count={1} />,
});

interface IFooterColumn {
  linkGroup: LinkGroups | null;
  dataUiTestId?: string;
}

const socialButtons = [
  {
    id: 'facebook',
    link: 'https://www.facebook.com/vanarama/',
    icon: <LogoFacebookV2 />,
  },
  {
    id: 'twitter',
    link: 'https://twitter.com/Vanarama',
    icon: <LogoTwitterV2 />,
  },
  {
    id: 'instagram',
    link: 'https://www.instagram.com/vanaramauk/',
    icon: <LogoInstagramV2 />,
  },
  {
    id: 'linkedin',
    link: 'https://www.linkedin.com/company/vanarama',
    icon: <LogoLinkedinV2 />,
  },
  {
    id: 'youtube',
    link: 'https://www.youtube.com/user/vanaramauk',
    icon: <LogoYouTubeV2 />,
  },
];

const FooterColumn: FC<IFooterColumn> = ({ linkGroup, dataUiTestId }) => {
  if (linkGroup) {
    return (
      <div
        className="footer--column"
        key={linkGroup.name || ''}
        data-uitestid={dataUiTestId}
      >
        <Heading
          color="medium"
          tag="span"
          size="small"
          dataUiTestId={`${dataUiTestId}_heading`}
        >
          {linkGroup.name}
        </Heading>
        {!(linkGroup.links?.length || linkGroup.linkGroups?.length) &&
          linkGroup.body && <ReactMarkdown source={linkGroup.body} />}
        {linkGroup.links && (
          <ul>
            {linkGroup.links?.map(link => (
              <li key={link?.text || ''}>
                <RouterLink
                  link={{ href: link?.url || '', label: link?.text || '' }}
                  classNames={{ color: 'white', size: 'small' }}
                  dataUiTestId={`${dataUiTestId}_link_${link?.text}`}
                />
              </li>
            ))}
          </ul>
        )}
        {linkGroup.linkGroups && (
          <>
            <ul>
              {linkGroup.linkGroups?.map(el => (
                <li className={el?.links ? '-mb-300' : ''} key={el?.name || ''}>
                  <Text size="small" color="medium" tag="span">
                    {el?.name}
                  </Text>{' '}
                  {el?.links?.map(link =>
                    link?.url ? (
                      <RouterLink
                        key={link?.text || ''}
                        link={{
                          href: link?.url || '',
                          label: link?.text || '',
                          linkType: LinkTypes.external,
                        }}
                        classNames={{ color: 'white', size: 'small' }}
                        withoutSilentReplacements
                      />
                    ) : (
                      <Text
                        size="small"
                        color="white"
                        tag="span"
                        key={link?.text || ''}
                      >
                        {link?.text}
                      </Text>
                    ),
                  )}
                </li>
              ))}
            </ul>

            <div className="footer--social-icons">
              {socialButtons.map(item => (
                <div
                  key={item.id}
                  className="button -white -small -outline -round"
                >
                  <a
                    href={item.link}
                    target="_blank"
                    className="button--inner"
                    data-uitestid={`${dataUiTestId}_${item.id}-social-link`}
                    rel="noreferrer"
                  >
                    <div>
                      <Icon size="small" color="white" icon={item.icon} />
                    </div>
                  </a>
                </div>
              ))}
            </div>
            <Logo assetName="bvrla" dataUiTestId={dataUiTestId} />
          </>
        )}
      </div>
    );
  }
  return <></>;
};

export default React.memo(FooterColumn);
