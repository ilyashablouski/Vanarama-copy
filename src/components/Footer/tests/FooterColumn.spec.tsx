import React from 'react';
import preloadAll from 'jest-next-dynamic';
import renderer from 'react-test-renderer';
import FooterColumn from '../FooterColumn';

describe('<FooterColumn />', () => {
  beforeEach(async () => {
    await preloadAll();
  });
  it('renders correctly with body', () => {
    const getComponent = () => {
      return renderer
        .create(
          <FooterColumn
            linkGroup={{
              body: 'test',
              links: null,
              name: 'GET IN TOUCH WITH US',
              linkGroups: null,
            }}
          />,
        )
        .toJSON();
    };

    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with links', () => {
    const getComponent = () => {
      return renderer
        .create(
          <FooterColumn
            linkGroup={{
              body: 'test',
              links: [
                {
                  text: 'Van Leasing',
                  url: 'https://vanarama.com/van-leasing.html',
                },
                {
                  text: 'Pickup Truck Leasing',
                  url: 'https://vanarama.com/pickup-truck-leasing.html',
                },
                {
                  text: 'Van Reviews',
                  url: 'https://vanarama.com/van-reviews.html',
                },
              ],
              name: 'VANS & TRUCKS',
              linkGroups: null,
            }}
          />,
        )
        .toJSON();
    };

    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });
});
