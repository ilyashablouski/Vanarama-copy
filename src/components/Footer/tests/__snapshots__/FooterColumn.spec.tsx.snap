// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterColumn /> renders correctly with body 1`] = `
<div
  className="footer--column"
  data-uitestid="footer_footer-column_ABOUT US"
>
  <span
    className="heading -medium -small"
    data-uitestid="footer_footer-column_ABOUT US_heading"
  >
    GET IN TOUCH WITH US
  </span>
  <p>
    test
  </p>
</div>
`;

exports[`<FooterColumn /> renders correctly with links 1`] = `
<div
  className="footer--column"
  data-uitestid="footer_footer-column_ABOUT US"
>
  <span
    className="heading -medium -small"
    data-uitestid="footer_footer-column_ABOUT US_heading"
  >
    VANS & TRUCKS
  </span>
  <ul>
    <li>
      <a
        className="link -white -small"
        data-testid="link"
        data-uitestid="footer_footer-column_ABOUT US_link_Van Leasing"
        href="https://vanarama.com/van-leasing.html"
        onClick={[Function]}
      >
        Van Leasing
      </a>
    </li>
    <li>
      <a
        className="link -white -small"
        data-testid="link"
        data-uitestid="footer_footer-column_ABOUT US_link_Pickup Truck Leasing"
        href="https://vanarama.com/pickup-truck-leasing.html"
        onClick={[Function]}
      >
        Pickup Truck Leasing
      </a>
    </li>
    <li>
      <a
        className="link -white -small"
        data-testid="link"
        data-uitestid="footer_footer-column_ABOUT US_link_Van Reviews"
        href="https://vanarama.com/van-reviews.html"
        onClick={[Function]}
      >
        Van Reviews
      </a>
    </li>
  </ul>
</div>
`;
